website/server/controllers/api-v3/auth.js

```diff
@@ -75,6 +75,10 @@ api.registerLocal = {
       email: {
         notEmpty: {errorMessage: res.t('missingEmail')},
         isEmail: {errorMessage: res.t('notAnEmail')},
+        matches: {
+          options: ['.+@(?!(?:(?:habitrpg\.com$)|(?:habitica\.com$))).+', 'i'],
```

Comment:
Not sure if that can be done with the schema, but I can look.
On Mon, Jun 13, 2016 at 10:38 PM Blade Barringer <notifications@github.com>
wrote:

> In website/server/controllers/api-v3/auth.js
> <https://github.com/HabitRPG/habitrpg/pull/7659#discussion_r66904749>:
>
> > @@ -75,6 +75,10 @@ api.registerLocal = {
> >        email: {
> >          notEmpty: {errorMessage: res.t('missingEmail')},
> >          isEmail: {errorMessage: res.t('notAnEmail')},
> > +        matches: {
> > +          options: ['.+@(?!(?:(?:habitrpg\.com$)|(?:habitica\.com$))).+', 'i'],
>
> This is a messy regex. Would prefer a custom validate method.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/HabitRPG/habitrpg/pull/7659/files/982e8f9e8be7b7a8650d4691a81341bfd05c1510#r66904749>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ALz_81m5BO4ASiVTzfAfAT-owX6nN-TFks5qLiI_gaJpZM4I06jn>
> .
>